### PR TITLE
Add setlocale() call (fix #1740)

### LINKED ISSUE #1740: Missing handling of locale-dependent format for strftime/strflocaltime and strptime
**Description**
Using a "it_IT.UTF-8" locale, the output of the strftime/strflocaltime function in jq differs from the output of the date command in the same environment

```
$ ls -l /etc/localtime
lrwxrwxrwx 1 root root 31 ott  3 12:38 /etc/localtime -> /usr/share/zoneinfo/Europe/Rome
$ locale -a
C
C.UTF-8
en_US.utf8
it_IT.utf8
POSIX
$ locale
LANG=it_IT.UTF-8
LANGUAGE=
LC_CTYPE="it_IT.UTF-8"
LC_NUMERIC="it_IT.UTF-8"
LC_TIME="it_IT.UTF-8"
LC_COLLATE="it_IT.UTF-8"
LC_MONETARY="it_IT.UTF-8"
LC_MESSAGES="it_IT.UTF-8"
LC_PAPER="it_IT.UTF-8"
LC_NAME="it_IT.UTF-8"
LC_ADDRESS="it_IT.UTF-8"
LC_TELEPHONE="it_IT.UTF-8"
LC_MEASUREMENT="it_IT.UTF-8"
LC_IDENTIFICATION="it_IT.UTF-8"
LC_ALL=it_IT.UTF-8
$ date 
gio 11 ott 2018, 15.51.21, CEST
$ jq -n 'now | strflocaltime("%a %d %b %Y at %H:%M:%S")'
"Thu 11 Oct 2018 at 15:51:38"
$ jq -n 'now | strftime("%a %d %b %Y at %H:%M:%S")'
"Thu 11 Oct 2018 at 13:52:00"
$ date +"%a %d %b %Y at %H:%M:%S"
gio 11 ott 2018 at 16:02:37
$ date +"%a %d %b %Y at %H:%M:%S" | jq -R 'strptime("%a %d %b %Y at %H:%M:%S")'
jq: error (at <stdin>:1): date "gio 11 ott 2018 at 16:04:20" does not match format "%a %d %b %Y at %H:%M:%S"
$ export LC_ALL=en_US.UTF-8
$ date +"%a %d %b %Y at %H:%M:%S"
Thu 11 Oct 2018 at 16:04:51
$ date +"%a %d %b %Y at %H:%M:%S" | jq -R 'strptime("%a %d %b %Y at %H:%M:%S")'
[
  2018,
  9,
  11,
  16,
  4,
  55,
  4,
  283
]
```

**To Reproduce**
Generate a non-English locale, export the appropriate variables and run the commands above.
**Expected behavior**
For jq to output/parse dates using the correct representation of names according to the current locale

**Environment (please complete the following information):**
 - OS: Ubuntu Linux 16.04
 - jq version: jq-1.6rc1-16-g341a5fc-dirty

**Additional context**
jq compiled from the latest tree
Using the version shipped with the distro (only with strftime) gives the same output.

### PR DESCRIPTION
**Pull request by nicowilliams**
Closes #1740

### CODE REVIEW

File: src/main.c [hunk 1]
```diff
@@ -319,6 +322,8 @@ int main(int argc, char* argv[]) {
   jv ARGS = jv_array(); /* positional arguments */
   jv program_arguments = jv_object(); /* named arguments */
 
+  (void) setlocale(LC_ALL, "");
```
Comment (emanuele6) [inline src/main.c:326]: `#ifdef HAVE_SETLOCALE`

**Comment by wader** [comment]
Could https://github.com/jqlang/jq/issues/1740#issuecomment-1708074625 be turned into some kind of shtest regression test?

**Comment by nicowilliams** [comment]
> Could [#1740 (comment)](https://github.com/jqlang/jq/issues/1740#issuecomment-1708074625) be turned into some kind of shtest regression test?

Sure, we could `locale -a` and pick a non-English locale.

**Comment by nicowilliams** [comment]
> Could [#1740 (comment)](https://github.com/jqlang/jq/issues/1740#issuecomment-1708074625) be turned into some kind of shtest regression test?

I've added a test!

**Comment by nicowilliams** [comment]
Working on the dist CI build...

**Comment by nicowilliams** [comment]
~Not sure why this change should break the `dist` CI build, or how to extract the test log.~

**Comment by nicowilliams** [comment]
Hmm, the test isn't quite right.

**Comment by nicowilliams** [comment]
image

**Pull request merged**